### PR TITLE
Merge two fread readers of hb_blob_create_from_file

### LINKED ISSUE #1063: Merge two fread readers of hb_blob_create_from_file
Simply, these two, [1](https://github.com/harfbuzz/harfbuzz/blob/941f2b85489b/src/hb-blob.cc#L502-L530) [2](https://github.com/harfbuzz/harfbuzz/blob/941f2b85489b/src/hb-blob.cc#L625-L638) having many things in common and should be merged (so we would have a `fread` reader without `ftell` at the end). I tried to do that initially but it revealed most of our bots are relying on `hb_blob_create_from_file`'s `fread` reader and don't have HAVE_MMAP which feels very wrong (and possibly shows just added `fread` reader, `_fread_to_end`, is doing something wrong also).

Filing this as a bug that I can't see any of these locally :/ I can try logging into the bots however but I think I should get some help from others.

### PR DESCRIPTION
**Pull request by ebraminio**
Fixes #1063.

And assign our Alpine CI bot to always use fallback reader.

### CODE REVIEW

File: src/hb-blob.cc [hunk 1]
```diff
@@ -621,32 +589,49 @@ hb_blob_create_from_file (const char *file_name)
...
+  if (unlikely (fp == nullptr)) goto fread_fail_without_close;
+
+  while (!feof (fp))
+  {
+    if (allocated - len < BUFSIZ)
+    {
+      allocated *= 2;
+      // Don't allocate more than 200MB, our mmap reader still
+      // can cover files like that but lets limit our fallback reader
+      if (unlikely (allocated > 200000000)) goto fread_fail;
```
Comment (ebraminio) [inline src/hb-blob.cc]: So no more fun with /dev/zero /dev/random /dev/urandom yes and friend :)

File: src/hb-blob.cc [hunk 2]
```diff
@@ -568,45 +538,43 @@ hb_blob_create_from_file (const char *file_name)
...
+
+  close (fd);
+
+  return hb_blob_create (file->contents, file->length, mm, (void *) file,
+			 (hb_destroy_func_t) _hb_mapped_file_destroy);
+
+fail:
+  close (fd);
+fail_without_close:
+  free (file);
```
Comment (ebraminio) [inline src/hb-blob.cc]: So this will always fallback on failures.

File: .circleci/config.yml [hunk 3]
```diff
@@ -40,14 +40,14 @@ jobs:
...
-  alpine-O3:
+  alpine-O3-NOMMAP:
     docker:
       - image: alpine
     steps:
       - checkout
       - run: apk update && apk add ragel make pkgconfig libtool autoconf automake gettext gcc g++ glib-dev freetype-dev cairo-dev
       # C??FLAGS are not needed for a regular build
-      - run: CFLAGS="-O3" CXXFLAGS="-O3" ./autogen.sh
+      - run: CFLAGS="-O3" CXXFLAGS="-O3 -DNOMMAPFILEREADER" ./autogen.sh
```
Comment (ebraminio) [inline .circleci/config.yml]: Is "NOMMAPFILEREADER" good?

File: src/hb-blob.cc (hunk 1 above)
Comment (ebraminio) [reply src/hb-blob.cc]: Is this a good limitation?

**Comment by ebraminio** [comment]
WDYT Behdad?

File: .circleci/config.yml (hunk 3 above)
Comment (behdad) [reply .circleci/config.yml]: I prefer always a `HB_` prefix.  I'd go for `HB_NOMMAP`.

File: src/hb-blob.cc (hunk 1 above)
Comment (behdad) [reply src/hb-blob.cc]: For multi-line comments please use C-style.  I personally always use C-style comments for, well, comments. And use C++-style for code that is commented out.

**Pull request merged**